zerver/tests/test_subs.py

```diff
@@ -823,6 +826,158 @@ def test_api_calls(self):
+            streams.append(stream)
+
+        result = self.client_post('/json/default_stream_groups',
+                                  {"group_name": group_name, "stream_names": ujson.dumps(stream_names)})
+        self.assert_json_success(result)
+
+        default_stream_groups = get_default_stream_groups_for_realm(realm)
+        self.assert_length(default_stream_groups, 1)
+        self.assertEqual(default_stream_groups[0].name, group_name)
+        self.assertEqual(list(default_stream_groups[0].streams.all()), streams)
+
+        new_stream_names = ["stream4", "stream5"]
+        new_streams = []
+        for new_stream_name in new_stream_names:
+            (new_stream, _) = create_stream_if_needed(realm, new_stream_name)
+            new_streams.append(new_stream)
+            streams.append(new_stream)
+
+        result = self.client_patch("/json/default_stream_groups",
+                                   {"group_name": group_name,
+                                    "stream_names": ujson.dumps(new_stream_names)})
+        self.assert_json_error(result, "Missing 'op' argument")
+
+        result = self.client_patch("/json/default_stream_groups",
+                                   {"group_name": group_name, "op": "invalid",
+                                    "stream_names": ujson.dumps(new_stream_names)})
+        self.assert_json_error(result, 'Nothing to do. Specify at least one of "add" or "delete".')
+
+        result = self.client_patch("/json/default_stream_groups",
+                                   {"group_name": "random group name", "op": "add",
+                                    "stream_names": ujson.dumps(new_stream_names)})
+        self.assert_json_error(result, "Default stream group random group name does not exist")
+
+        do_add_default_stream(new_streams[0])
+        result = self.client_patch('/json/default_stream_groups',
+                                   {"group_name": group_name, "op": "add", "stream_names": ujson.dumps(new_stream_names)})
+        self.assert_json_error(result, "stream4 is a default stream and cannot be added to group1")
+
+        do_remove_default_stream(new_streams[0])
+        result = self.client_patch("/json/default_stream_groups",
+                                   {"group_name": group_name, "op": "add",
+                                    "stream_names": ujson.dumps(new_stream_names)})
+        default_stream_groups = get_default_stream_groups_for_realm(realm)
+        self.assert_length(default_stream_groups, 1)
+        self.assertEqual(default_stream_groups[0].name, group_name)
+        self.assertEqual(list(default_stream_groups[0].streams.all()), streams)
+
+        result = self.client_patch('/json/default_stream_groups',
+                                   {"group_name": group_name, "op": "add", "stream_names": ujson.dumps(new_stream_names)})
+        self.assert_json_error(result, "Stream stream4 is already present in  default stream group group1")
```

Comment:
the double-space here probably means the error string has 2 spcaes and should be fixed.